src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ServiceDescriptionService.java

```diff
@@ -290,14 +291,13 @@ public ServiceDescriptionType addOpenapi3ServiceDescription(ClientId clientId, S
             ServiceCodeAlreadyExistsException {
         verifyAuthority("ADD_OPENAPI3");
 
-        // Parse openapi definition
-        OpenApiParser.Result result = null;
-        try {
-            result = openApiParser.parse(url);
-        } catch (OpenApiParser.ParsingException e) {
-            throw e;
+        if (serviceCode == null) {
+            throw new BadRequestException("Missing ServiceCode");
```

Comment:
The `BadRequestException` should be moved into the controller. 

This particular case is somewhat unusual because the serviceCode param should actually be validated already in the API layer but we can't do that because we have a shared endpoint for adding WSDL's and OPENAPI/REST types.

Reply:
There is also another one [here](https://github.com/nordic-institute/X-Road/pull/309/commits/6ac518dbfb9a83dd12384cc9b3192012a662ae7b#diff-d5a64e5ee379cb7ea245a3fc7a9247e0R406-R408)